File: RetailStation.Entities/Common/Finance/GeneralAccounts/ReceiptLedgerModel.cs
using RetailStation.Entities.Models;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace RetailStation.Entities.Common.Finance.GeneralAccounts
{
    public class ReceiptLedgerModel : CreatorModel
    {
        public int? ReceiptLedgerId { get; set; }
        public int StartReceiptNumber { get; set; }
        public int FinancialPeriodId { get; set; }
        public int ReceiptLedgerTypeId { get; set; }
        public int PaymentTypeId { get; set; }
        public bool IsActive { get; set; }
        public bool IsLocked { get; set; }
        public string Code { get; set; }
        public string NameAR { get; set; }
        public string NameEN { get; set; }
        public string Notes { get; set; }


        public string FinancialPeriodNameEN { get; set; }
        public string FinancialPeriodNameAR { get; set; }
        public string ReceiptLedgerTypeNameEN { get; set; }
        public string ReceiptLedgerTypeNameAR { get; set; }
        public string PaymentTypeNameEN { get; set; }
        public string PaymentTypeNameAR { get; set; }

        public int? TotalCount { get; set; }
    }
}

File: RetailStation.Angular/src/app/components/GeneralAccounts/models/ReceiptLedgerModel.ts
export interface ReceiptLedgerModel {
    receiptLedgerId: number | null;
    startReceiptNumber: number;
    financialPeriodId: number;
    receiptLedgerTypeId: number;
    paymentTypeId: number;
    isActive: boolean;
    isLocked: boolean;
    code: string;
    nameAR: string;
    nameEN: string;
    notes: string;

    financialPeriodNameEN: string;
    financialPeriodNameAR: string;
    receiptLedgerTypeNameEN: string;
    receiptLedgerTypeNameAR: string;
    paymentTypeNameEN: string;
    paymentTypeNameAR: string;
    totalCount: number | null;
}
